File: .sdk/tm/php/utility/ResultBody.php
<?php
declare(strict_types=1);

// HongKongCsdi SDK utility: result_body

class HongKongCsdiResultBody
{
    public static function call(HongKongCsdiContext $ctx): ?HongKongCsdiResult
    {
        $response = $ctx->response;
        $result = $ctx->result;
        if ($result && $response && $response->json_func && $response->body) {
            $result->body = ($response->json_func)();
        }
        return $result;
    }
}

File: go/utility/result_body.go
package utility

import "github.com/voxgig-sdk/hong-kong-csdi-sdk/core"

func resultBodyUtil(ctx *core.Context) *core.Result {
	response := ctx.Response
	result := ctx.Result

	if result != nil {
		if response != nil && response.JsonFunc != nil && response.Body != nil {
			json := response.JsonFunc()
			result.Body = json
		}
	}

	return result
}
